core/rio/hdt/src/main/java/org/eclipse/rdf4j/rio/hdt/HDTParser.java

```diff
@@ -0,0 +1,311 @@
+	// buffer for input stream
+	private final static int BUFLEN = 16384;
+
+	private Map<String, String> globalProps;
+	private Map<String, String> headerProps;
+	private byte[] headerData;
+
+	/**
+	 * Creates a new HDTParser that will use a {@link SimpleValueFactory} to create RDF model objects.
+	 */
+	public HDTParser() {
+		super();
+	}
+
+	/**
+	 * Creates a new HDTParser that will use the supplied ValueFactory to create RDF model objects.
+	 *
+	 * @param valueFactory A ValueFactory.
+	 */
+	public HDTParser(ValueFactory valueFactory) {
+		super(valueFactory);
+	}
+
+	@Override
+	public RDFFormat getRDFFormat() {
+		return RDFFormat.HDT;
+	}
+
+	@Override
+	public Collection<RioSetting<?>> getSupportedSettings() {
+		Set<RioSetting<?>> result = new HashSet<>();
+		return result;
+	}
+
+	/**
+	 * Get global properties
+	 * 
+	 * @return key,value map
+	 */
+	public Map<String, String> getGlobalProperties() {
+		return globalProps;
+	}
+
+	/**
+	 * Get header properties
+	 * 
+	 * @return key,value map
+	 */
+	public Map<String, String> getHeaderProperties() {
+		return headerProps;
+	}
+
+	/**
+	 * Get header data as raw byte stream. This often includes a series of triples (in N-Triple) with additional
+	 * metadata and statistics.
+	 * 
+	 * @return bytes
+	 */
+	public byte[] getHeaderData() {
+		return headerData;
+	}
+
+	/**
+	 * Implementation of the <tt>parse(InputStream, String)</tt> method defined in the RDFParser interface.
+	 *
+	 * @param in      The InputStream from which to read the data, must not be <tt>null</tt>.
+	 * @param baseURI The URI associated with the data in the InputStream, must not be <tt>null</tt>.
+	 * @throws IOException              If an I/O error occurred while data was read from the InputStream.
+	 * @throws RDFParseException        If the parser has found an unrecoverable parse error.
+	 * @throws RDFHandlerException      If the configured statement handler encountered an unrecoverable error.
+	 * @throws IllegalArgumentException If the supplied input stream or base URI is <tt>null</tt>.
+	 */
+	@Override
+	public synchronized void parse(InputStream in, String baseURI)
+			throws IOException, RDFParseException, RDFHandlerException {
+		if (in == null) {
+			throw new IllegalArgumentException("Input stream must not be 'null'");
+		}
+
+		if (in instanceof FileInputStream) {
+			// "TODO: use more optimized way to parse the file, eg. filechannel / membuffer"
+		}
+
+		HDTDictionarySection shared = null;
+		HDTDictionarySection subjects = null;
+		HDTDictionarySection predicates = null;
+		HDTDictionarySection objects = null;
+		HDTTriplesSection section = null;
+
+		// not using try-with-resources, since the counter is needed in the catch clause (JDK8)
+		CountingInputStream bis = new CountingInputStream(new BufferedInputStream(in, BUFLEN));
+		try {
+			reportLocation(0, -1);
+			HDTGlobal global = new HDTGlobal();
+			global.parse(bis);
+			globalProps = global.getProperties();
```

Comment:
This seems to be the only reference to this field other than its unused accessor. Is it necessary to make this a field and not a local variable inside of this method?